CreateUser/handler.ts

```diff
@@ -0,0 +1,192 @@
+import { Context } from "@azure/functions";
+import { GraphRbacManagementClient } from "@azure/graph";
+import * as msRestNodeAuth from "@azure/ms-rest-nodeauth";
+import * as express from "express";
+import { toError } from "fp-ts/lib/Either";
+import { fromEither, TaskEither, tryCatch } from "fp-ts/lib/TaskEither";
+import { ServiceModel } from "io-functions-commons/dist/src/models/service";
+import {
+  AzureApiAuthMiddleware,
+  IAzureApiAuthorization,
+  UserGroup
+} from "io-functions-commons/dist/src/utils/middlewares/azure_api_auth";
+import {
+  AzureUserAttributesMiddleware,
+  IAzureUserAttributes
+} from "io-functions-commons/dist/src/utils/middlewares/azure_user_attributes";
+import {
+  ClientIp,
+  ClientIpMiddleware
+} from "io-functions-commons/dist/src/utils/middlewares/client_ip_middleware";
+import { ContextMiddleware } from "io-functions-commons/dist/src/utils/middlewares/context_middleware";
+import { RequiredBodyPayloadMiddleware } from "io-functions-commons/dist/src/utils/middlewares/required_body_payload";
+import { withRequestMiddlewares } from "io-functions-commons/dist/src/utils/request_middleware";
+import {
+  checkSourceIpForHandler,
+  clientIPAndCidrTuple as ipTuple
+} from "io-functions-commons/dist/src/utils/source_ip_check";
+import { wrapRequestHandler } from "italia-ts-commons/lib/request_middleware";
+import {
+  IResponseErrorInternal,
+  IResponseSuccessJson,
+  ResponseSuccessJson
+} from "italia-ts-commons/lib/responses";
+import * as randomString from "randomstring";
+import { ulid } from "ulid";
+import { User } from "../generated/definitions/User";
+import { UserPayload } from "../generated/definitions/UserPayload";
+import {
+  getApiClient,
+  IAzureApimConfig,
+  IServicePrincipalCreds
+} from "../utils/apim";
+import { userContractToApiUser } from "../utils/conversions";
+import { genericInternalErrorHandler } from "../utils/errorHandler";
+
+type ICreateUserHandler = (
+  context: Context,
+  auth: IAzureApiAuthorization,
+  clientIp: ClientIp,
+  userAttributes: IAzureUserAttributes,
+  userPayload: UserPayload
+) => Promise<IResponseSuccessJson<User> | IResponseErrorInternal>;
+
+function getGraphRbacManagementClient(
+  adb2cCreds: IServicePrincipalCreds
+): TaskEither<Error, GraphRbacManagementClient> {
+  return tryCatch(
+    () =>
+      msRestNodeAuth.loginWithServicePrincipalSecret(
+        adb2cCreds.clientId,
+        adb2cCreds.secret,
+        adb2cCreds.tenantId,
+        { tokenAudience: "graph" }
+      ),
+    toError
+  ).map(
+    credentials =>
+      // the below cast to `any` is needed to prevent build errors
```

Comment:
this should be solved using 4.x version of azure graph package:

https://github.com/gunzip/adb2c-apim-migrate/blob/master/package.json#L8